inlong-dashboard/src/plugins/sinks/defaults/MySQL.ts

```diff
@@ -47,7 +47,7 @@ const fieldTypesConf = {
   TIME: () => '',
   DATETIME: () => '',
   CHAR: (m, d) => (1 <= m && m <= 255 ? '' : '1<=M<=255'),
-  VARCHAR: (m, d) => (1 <= m && m <= 255 ? '' : '1<=M<=255'),
+  VARCHAR: (m, d) => (1 <= m && m <= 16383 ? '' : '1<=M<=16383'),
```

Comment:
It seems that the maximum length of MySQL's varchar type is 65535?

Reply:
Excuse me, how do you get the result of 16383 here?

Reply:
> Excuse me, how do you get the result of 16383 here?
> 
> If you consider the utf8mb4 character set, it takes up to 4 bytes to represent 1 character, so 65535 / 4 = 16383, then the maximum length of the char above is 255, which is also inaccurate.

What is the maximum length of the varchar field in MySQL version 5.7? Under different character sets, the disk space occupied by a character varies, so:

Under the Litan1 character set, varchar can store up to 65535-2-1=65532 characters. If it is a non empty varchar field, it can store 65535-2=65533 characters.

Under the utf8 character set, varchar can store up to floor ((65535-2-1)/3)=21844 characters. If it is a non empty varchar field, it can store floor ((65535-2)/3)=21844, and after rounding down, it is also 21844 characters

Under the utf8mb4 character set, varchar can store up to floor (65535-2-1)/4)=16383 characters. If it is a non empty varchar field, it can store 16383 characters as well.
![13d831694265891853e93a00eaa2e53](https://github.com/apache/inlong/assets/54351417/dc57cacb-561a-4fd7-8254-a0db715f2605)

Reply:
So, do we need to limit the length of the CHAR type to `(255-2-1)/4=63`?

Reply:
> So, do we need to limit the length of the CHAR type to `(255-2-1)/4=63`?

In MySQL, char is a data type used to store fixed length strings. The maximum length of a char type is 255 characters, and smaller lengths can also be set. Among them, a fixed length string refers to a fixed length that is not affected by the stored data.

Reply:
I found that in MySQL 8.0.30 on macOS, the default charset is utf8mb4, in this case, the max length of the CHAR type is 255, and the max length of the VARCHAR type is 16383, so this PR is good to me.

Charset:

![image](https://github.com/apache/inlong/assets/31994335/e4514b20-1af0-48ec-b189-ce8913f838f9)

CHAR type:

<img width="827" alt="image" src="https://github.com/apache/inlong/assets/31994335/48b19e91-8184-489c-934d-c93663857162">

VARCHAR type:
<img width="878" alt="image" src="https://github.com/apache/inlong/assets/31994335/31983eea-5cd3-48fb-adcf-c789b902218b">